### PR TITLE
vscode debug config for unit tests

### PR DESCRIPTION
**Pull request by ostreifel**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,12 +1,17 @@
 {
     "compilerOptions": {
-        "module": "amd",
+        "module": "umd",
```
Comment (cschleiden) [inline tsconfig.json]: why this change? Since VSTS uses AMD I'd like to keep the module format consistent

File: tsconfig.json [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
         "sourceMap": false,
         "rootDir": "./scripts"
     },
     "filesGlob": [
         "./scripts/**/*.ts",
         "!**/*.tests.ts",
         "./typings/index.d.ts"
+    ],
+    "files": [
```
Comment (cschleiden) [inline tsconfig.json]: I think these should not be here, any tool that supports `filesGlob` should be able to expand them

File: .gitignore [hunk 3]
```diff
@@ -2,4 +2,6 @@ node_modules
 scripts/*.js
 typings
 dist
-build
\ No newline at end of file
+build
+test
+.baseDir.ts
```
Comment (cschleiden) [inline .gitignore:7]: when is this added?

**Review by cschleiden — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 1 above)
Comment (ostreifel) [reply tsconfig.json]: Node requires special configuration to use amd. So I went with the compromise module configuration. Alternatively we could have a separate tsconfig file for tests to use commonjs module configuration

**Comment by ostreifel** [comment]
Updated to have tests use a separate tsconfig file so that ./build uses amd and ./test uses umd (Needs to be both amd compatible for grunt test and commonjs compatible when debugging with node). Also removed duplicate type definitions that break the build when the '@types' directory is added in node_modules
